### PR TITLE
Auto layout fixes

### PR DESCRIPTION
**Pull request by pec1985**

### CODE REVIEW

**Comment by pec1985** [comment]
@cheekiatng **merge me!**

**Comment by pec1985** [comment]
This is the command line I'm using for testing through the terminal:
```
cd {repo}/iphone/layout
xcodebuild -scheme layout -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPad Air 2,OS=9.2' test
```

File: iphone/Classes/TiUIScrollableView.m [hunk 1]
```diff
@@ -152,24 +172,25 @@ -(void)layoutSubviews
 -(void)addSubview:(nonnull UIView *)view
 {
     WRAP_TI_VIEW(view)
-    [_contentView addSubview:wrapperView];
+    [wrapperView setBackgroundColor:[UIColor redColor]];
```
Comment (hansemannn) [inline iphone/Classes/TiUIScrollableView.m]: Shouldn't this be `[UIColor clearColor]` or `nil` ?

File: iphone/Classes/TiUIScrollableView.m [hunk 2]
```diff
@@ -212,20 +233,14 @@ -(UIScrollView*)scrollview
...
-        [self setDefaultHeight:TiDimensionFromObject(@"FILL")];
-        [self setDefaultWidth:TiDimensionFromObject(@"FILL")];
-        
-        _contentView = [[UIView alloc] init];
-        [_contentView setTranslatesAutoresizingMaskIntoConstraints:NO];
-        
+	{        
         _scrollView = [[UIScrollView alloc] init];
-        [_scrollView setDelegate:self];
+            [_scrollView setDelegate:self];
```
Comment (hansemannn) [inline iphone/Classes/TiUIScrollableView.m]: Indentation

File: iphone/Classes/TiUIScrollableView.m [hunk 3]
```diff
@@ -715,9 +742,10 @@ -(void)pageControlTouched:(id)sender
 	[scrollview setContentOffset:CGPointMake([self bounds].size.width * pageNum, 0) animated:YES];
 	handlingPageControlEvent = YES;
 	
+#ifndef TI_USE_AUTOLAYOUT
 	currentPage = pageNum;
```
Comment (hansemannn) [inline iphone/Classes/TiUIScrollableView.m]: Keeping track of the current page should be done even if AL is enabled

File: iphone/Classes/TiUISwitch.h [hunk 4]
```diff
@@ -10,6 +10,8 @@
 @interface TiUISwitch : TiUIView<LayoutAutosizing> {
 @private
 	UISwitch *switchView;
+    BOOL firstInit;
+    BOOL animated;
```
Comment (hansemannn) [inline iphone/Classes/TiUISwitch.h]: Indentation

File: iphone/Classes/TiUITableViewRowProxy.h [hunk 5]
```diff
@@ -24,7 +24,11 @@
 	TiDimension topCap;
 	BOOL configuredChildren;
 	int dirtyRowFlags;
-	UIView * rowContainerView;
+#ifdef TI_USE_AUTOLAYOUT
+    TiLayoutView * rowContainerView;
+#else
+    UIView * rowContainerView;
```
Comment (hansemannn) [inline iphone/Classes/TiUITableViewRowProxy.h]: Indentation..damn those legacy :-1:

File: iphone/Classes/TiUITextWidget.m [hunk 6]
```diff
@@ -18,6 +18,15 @@
 
 @implementation TiUITextWidget
 
+#ifdef TI_USE_AUTOLAYOUT
+-(void)initializeTiLayoutView
+{
+    [super initializeTiLayoutView];
+    [self setDefaultHeight:TiDimensionAutoSize];
+    [self setDefaultWidth:TiDimensionAutoFill];
```
Comment (hansemannn) [inline iphone/Classes/TiUITextWidget.m]: /cc @FokkeZB, didn't we have a ticket for that behavior?

Comment (FokkeZB) [inline iphone/Classes/TiUITextWidget.m]: Yes, mentioned that we should do this for ProgressBar and TextField here:
https://jira.appcelerator.org/browse/TIMOB-20406

**Comment by cheekiatng** [comment]
CR and FT passed.

**Pull request merged**